examples/setup.yaml

```diff
@@ -11,575 +11,760 @@
                             type: string
-                          to:
-                            description: Remapped Prometheus target label. Defaults to the same name as `From`.
+                          for:
+                            description: The duration to wait before a firing alert
+                              produced by this rule is sent to Alertmanager. Only
+                              valid if `alert` is set.
                             type: string
-            status:
-              description: Most recently observed status of the resource.
-              type: object
-              properties:
-                conditions:
-                  description: Represents the latest available observations of a podmonitor's current state.
-                  type: array
-                  items:
-                    description: MonitoringCondition describes a condition of a PodMonitoring.
-                    type: object
-                    required:
-                      - status
-                      - type
-                    properties:
-                      lastTransitionTime:
-                        description: Last time the condition transitioned from one status to another.
-                        type: string
-                        format: date-time
-                      lastUpdateTime:
-                        description: The last time this condition was updated.
-                        type: string
-                        format: date-time
-                      message:
-                        description: A human-readable message indicating details about the transition.
-                        type: string
-                      reason:
-                        description: The reason for the condition's last transition.
-                        type: string
-                      status:
-                        description: Status of the condition, one of True, False, Unknown.
-                        type: string
-                      type:
-                        description: MonitoringConditionType is the type of MonitoringCondition.
-                        type: string
-                observedGeneration:
-                  description: The generation observed by the controller.
-                  type: integer
-                  format: int64
-  scope: Namespaced
+                          labels:
+                            additionalProperties:
+                              type: string
+                            description: A set of labels to attach to the result of
+                              the query expression.
+                            type: object
+                          record:
+                            description: Record the result of the expression to this
+                              metric name. Only one of `record` and `alert` must be
+                              set.
+                            type: string
+                        required:
```

Comment:
The required lists are still edited manually on each generation?

Reply:
Discussed offline - consensus is that this is controlled via `"omitempty"` json tag in types.go - e.g. https://github.com/GoogleCloudPlatform/prometheus-engine/blob/55ae264b745f71b5fa28cb4daabd09d432c81aa0/pkg/operator/apis/monitoring/v1alpha1/types.go#L598